Add instructions for udev rules on "host" machine and lxc-android PR

Herrie82 opened the pull request:
This gets rid of the to-do's :)

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -7,10 +7,12 @@ Now that you're logged in, you must create some udev rules to allow some tests t
 
```
Please clarify that the above paragraph is for ON the device and please remove the . from codename.::

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -7,10 +7,12 @@ Now that you're logged in, you must create some udev rules to allow some tests t
 Now, reboot the device.
 
-.. todo::
+Alternatively if you want to create the file with the udev rules on your "host" machine you can run the following command inside your halium folder, replacing [codename] with your device's codename.::
```
remove the dot from codename.::

doniks replied:
And I think (no native speaker) a comma after Alternatively

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -7,10 +7,12 @@ Now that you're logged in, you must create some udev rules to allow some tests t
 
-    Update these instructions to update udev rules on "host" machine. The ueventd*rc files are already in $OUT/root/ , so it needs updating for path and then instead of writing it to /etc/udev/rules.d/70-codename.rules, writing it to local file and then adb pushing it while in recovery.
+    cat /out/target/product/[codename]/root/ueventd*.rc|grep ^/dev|sed -e 's/^\/dev\///'|awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' > 70-[codename].rules
 
```
lets add spaces around the pipe characters `|` to make it a bit more readable. Also above for on-device please.

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -7,10 +7,12 @@ Now that you're logged in, you must create some udev rules to allow some tests t
-    Add information about adding udev rules in the lxc-android repository.
+    adb push 70-[codename].rules /data/halium-rootfs/etc/udev/rules.d/
+
+In other for other users and devs to be able to use the udev rules as well, it's necessary to send a pull request containing the new file to the lxc-android repository. Similar to: https://github.com/Halium/lxc-android/pull/12/files
```
typo? "In other for other"

doniks replied:
I'd suggest to remove "and devs"

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -7,10 +7,12 @@ Now that you're logged in, you must create some udev rules to allow some tests t
+    cat /out/target/product/[codename]/root/ueventd*.rc|grep ^/dev|sed -e 's/^\/dev\///'|awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' > 70-[codename].rules
 
-.. todo::
+You can then use adb push from recovery to push the file to /data/halium-rootfs/etc/udev/rules.d/ i.e.::
```
put backticks around `adb push` and the path

Herrie82 commented:
@doniks @JBBgameich I updated based on your comments, let me know if there's anything else to change. I've squashed everything into a single commit as well.

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -1,16 +1,24 @@
 Add udev rules
 ==============
```
Thanks, I think it can still be clearer

Please add three subsections: on device, on host, send upstream, i.e.:

Add udev rules
In order to allow some tests yadayadayada ...

On the device
Log in ... 

On the host
Alternatively, ...

Send upstream
In order for others ....

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -1,16 +1,24 @@
+    cat /out/target/product/[codename]/root/ueventd*.rc | grep ^/dev | sed -e 's/^\/dev\///' | awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' > 70-[codename].rules
 
-.. todo::
+You can then use either SSH or 'adb push' to push the file to the right location:
```
No, that's the wrong character, check other pages for how to mark up inline code. Should be ``ssh`` or ``adb push``

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -1,16 +1,24 @@
+You can then use either SSH or 'adb push' to push the file to the right location:
 
-    Add information about adding udev rules in the lxc-android repository.
+SSH:: 
```
Explain what to do here. Smt like: Boot into the rootfs and then

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -1,16 +1,24 @@
-    Add information about adding udev rules in the lxc-android repository.
+SSH:: 
+
+    cat 70-[codename].rules | ssh 10.15.19.82 'cat > /tmp/plan.md'
```
Now that I re-read this ... aren't we missing something here? I got the rules in ``/tmp/plan.md`` what's next?

doniks commented on porting/debug-build/udev.rst:
```diff
@@ -1,16 +1,24 @@
+Alternatively, if you want to create the file with the udev rules on your "host" machine you can run the following command inside your halium folder, replacing [codename] with your device's codename::
 
-    Update these instructions to update udev rules on "host" machine. The ueventd*rc files are already in $OUT/root/ , so it needs updating for path and then instead of writing it to /etc/udev/rules.d/70-codename.rules, writing it to local file and then adb pushing it while in recovery.
+    cat /out/target/product/[codename]/root/ueventd*.rc | grep ^/dev | sed -e 's/^\/dev\///' | awk '{printf "ACTION==\"add\", KERNEL==\"%s\", OWNER=\"%s\", GROUP=\"%s\", MODE=\"%s\"\n",$1,$3,$4,$2}' | sed -e 's/\r//' > 70-[codename].rules
```
that won't work. It's not in the root folder ``/out/...`` so definitely get rid of the leading /
Actually, why not update the text to

run the following command inside the output folder ``out/target/prodcut/[codename]/`` 

and then simplify the command

doniks commented:
FYI others, discussion here became too messy and stalled. Mailed a patch to @Herrie82

Herrie82 commented:
@doniks Done. Let me know if you want things squashed.

doniks commented:
ugh. damn. one more copy/paste mistake. sorry if that was in the patch I sent you. 
Now there is two times "There are 2 different ways to do this:"

Herrie82 commented:
@doniks Sorry for the delay this has now been fixed. If you want me to squash stuff let me know. But I think GH web interface offers squash now too ;)
